python3Packages.nix-prefetch-github: 6.0.1 -> 7.0.0

seppeljordan opened the pull request:
###### Description of changes

nix-prefetch-github was update to version 7.0.0.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

seppeljordan committed:
python3Packages.nix-prefetch-github: 6.0.1 -> 7.0.0

tjni commented:
Looks good! There are a few dozen usages of `nix-prefetch-github` in update scripts -- could those be updated in this PR to use the new `hash` field?

seppeljordan commented:
Do I need to do anything?

tjni commented:
I added some commits to do this in https://github.com/seppeljordan/nixpkgs/pull/1. Can you review and merge into this PR if it looks good to you?

tjni committed:
docs/contributing: fix "ha256" -> "sha256" typo

tjni committed:
psst: adapt update script to new nix-prefetch-github

tjni committed:
squeezelite: adapt update script to new nix-prefetch-github

tjni committed:
solana-validator: adapt update script to new nix-prefetch-github

The update script is also already broken and needs:

1. Add missing packages to the shebang.
2. Retrieve cargo hash from the proper package.

We also take the opportunity to switch ot using the newer "hash" and
"cargoHash" properties.

tjni committed:
emacsPackages.tsc: adapt update script to new nix-prefetch-github

The update script seems to have been broken already. More work needs to
be done to fix it. In the meantime, this PR also takes the opportunity
to switch to using newer "hash" and "cargoHash" attributes.

tjni committed:
vscode-extensions.vadimcn.vscode-lldb: adapt update script to new nix-prefetch-github

tjni committed:
retroarch: format hashes using new nix-prefetch-github output

tjni committed:
tandoor-recipes, tandoor-recipes.frontend: adapt update script to new nix-prefetch-github

We also take the opportunity also to migrate from "sha256" and
"yarnSha256" attributes to "hash" and "yarnHash" attributes.

tjni committed:
element-desktop, element-web: adapt update script to new nix-prefetch-github

tjni committed:
zammad: use new nix-prefetch-github output

tjni committed:
woodpecker-server, woodpecker-server.woodpecker-frontend: adapt update script to new nix-prefetch-github

At the same time, we opportunistically switch to using the newer "hash"
attribute.

tjni committed:
netlify-cli: adapt update script to new nix-prefetch-github

tjni committed:
pnpm-lock-export: adapt update script to new nix-prefetch-github

tjni committed:
jellyseerr: adapt update script to new nix-prefetch-github

tjni committed:
mastodon: adapt update script to new nix-prefetch-github

tjni committed:
matrix-appservice-discord: adapt update script to new nix-prefetch-github

tjni committed:
matrix-appservice-slack: adapt update script to new nix-prefetch-github

tjni committed:
matrix-hookshot: adapt update script to new nix-prefetch-github

tjni committed:
memos: adapt update script to new nix-prefetch-github

tjni committed:
lemmy-server, lemmy-ui: adapt update script to new nix-prefetch-github

tjni committed:
plausible: adapt update script to new nix-prefetch-github

Take the opportunity to switch to SRI hashes.

tjni committed:
treewide: remove unused nix-prefetch-github from shebangs

tjni commented on pkgs/servers/web-apps/lemmy/pin.json:
```diff
@@ -1,7 +1,7 @@
+  "serverHash": "sha256-jYbrbIRyXo2G113ReG32oZ56ed2FEB/ZBcqYAxoxzGQ=",
+  "serverCargoHash": "sha256-7DNMNPSjzYY45DlR6Eo2q6QdwrMrRb51cFOnXfOuub0=",
+  "uiHash": "sha256-tc7fGA4okIv+3kq5t6I+EN+owdekCgAdk0EtkDgodIU=",
+  "uiYarnDepshash": "sha256-/T1PQ0XsGY82/RlLq94hJgv8Yx1bIf2gcklhgtPu1RM="
```
Just a small typo here: `uiYarnDepshash` => `uiYarnDepsHash`.

seppeljordan replied:
Thanks. I fixed it.

tjni reviewed (commented):
Thank you for merging everything!

tjni commented:
We're a bit unlucky: https://github.com/NixOS/nixpkgs/pull/242130 introduced a conflict as well.

seppeljordan commented:
@tjni I addressed the conflict.

tjni commented:
Result of `nixpkgs-review pr 242092` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>solana-validator</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>element-desktop</li>
    <li>element-desktop-wayland</li>
    <li>emacsPackages.evil-textobj-tree-sitter</li>
    <li>emacsPackages.evil-tree-edit</li>
    <li>emacsPackages.helm-tree-sitter</li>
    <li>emacsPackages.pygn-mode</li>
    <li>emacsPackages.symex</li>
    <li>emacsPackages.tree-sitter</li>
    <li>emacsPackages.tree-sitter-ess-r</li>
    <li>emacsPackages.tree-sitter-indent</li>
    <li>emacsPackages.tree-sitter-ispell</li>
    <li>emacsPackages.tree-sitter-langs</li>
    <li>emacsPackages.tsc</li>
    <li>lemmy-server</li>
    <li>nix-prefetch-github (python310Packages.nix-prefetch-github)</li>
    <li>nix-prefetch-github.dist (python310Packages.nix-prefetch-github.dist)</li>
    <li>python311Packages.nix-prefetch-github</li>
    <li>python311Packages.nix-prefetch-github.dist</li>
    <li>schildichat-desktop</li>
    <li>schildichat-desktop-wayland</li>
  </ul>
</details>

tjni commented:
We have another (more complicated) merge conflict in https://github.com/NixOS/nixpkgs/pull/242979/. Do you mind if I take over this PR? It is taking awhile to get committed, and I feel bad that you have to deal with all of these conflicts related to what I've asked you to push.

seppeljordan commented:
@tjni sure, go ahead.

tjni commented:
Continued in https://github.com/NixOS/nixpkgs/pull/243018.
